### DISCUSSION TITLE
Problems after pushing the project on GitHub

### DISCUSSION THREAD

**Discussion by arbdev95** [Cyfrin/foundry-full-course-cu#2919] [Q&A]
Hello,

after i push the project on github and do "code" to refresh, i see my project, but with a red "x" near the title, it says "CI / Foundry project (push) " and after i go in details it says "Error: Process completed with exit code 1.", why is that and how do i solve this problem?

best regards

**Comment by EngrPips** [comment]
Hello @arbdev95, I am unsure what went wrong, but did you carefully follow Patrick's instructions on how to push your code to GitHub?

**Comment by LordAdhaar** [comment]
It could be the foundry workflow that runs by default that could be failing. If your code is getting pushed and you can see it in your repo, you don't have anything to worry about.

**Comment by EngrPips** [comment]
Interesting

**Comment by AnaAse** [comment]
Same here, went though all steps.. I see now a repository on my Github BUT it is showing a red X

15 files changed
+618
-90
lines changed


<img width="896" alt="Screenshot 2025-01-28 at 14 34 38" src="https://github.com/user-attachments/assets/f541cf72-5ac5-4f7a-bfa8-ae0f78a3ad26" />
<img width="992" alt="Screenshot 2025-01-28 at 19 10 26" src="https://github.com/user-attachments/assets/45673a41-f7d6-470b-bf01-aa07cfb15f3f" />

**Reply by EngrPips** [reply]
If all your code is pushed properly then you shouldn't sweat over this

**Reply by AnaAse** [reply]
AI is suggesting to run: forge fmt
My concern is if that will change all my files format and why. I followed all  the lessons and pushed to Github the same files. I did not happen to hear Patrick saying anything about forge fmt and Im afraid of running it without further advise.

**Reply by EngrPips** [reply]
Running `forge fmt` will do you no harm. You were able to push all the code to Github.

**Reply by AnaAse** [reply]
Is there a list showing which files & folders were uploaded to Github by Patrick?

I get this:
Failing tests:
Encountered 1 failing test in test/unit/ZkSyncDevOps.t.sol:ZkSyncDevOps
[FAIL: FoundryZkSyncChecker__UnknownFoundryVersion()] testZkSyncFoundryFails() (gas: 13398)

Im wondering if the whole zksync was part of the push or not and how to fix this UnknownFoundryVersion

**Reply by EngrPips** [reply]
To test stuff on ZKSync you often need some extra set-up or command flags did you take those into account?

**Reply by satyaorz** [reply]
i had the same problem, i don't remember what exactly but even if you keep up with Patrick in the video and code along with him, there are some changes that u need to do so it can be pushed to github, you can just look to the github code of the fundme project and make some adjustment then try to push it to github. that might work.

**Reply by AnaAse** [reply]
Hurray!... I removed the function testZkSyncFoundryFails from ZkSyncDevOps and also did some formatting again, forge fmt. No more red X, let´s go!

**Reply by EngrPips** [reply]
Awesome, Keep crushing it.

**Comment by mariofilbert** [comment]
Try adding `forge fmt` in the github workflow folder, specifically inside the .yml file

![Screenshot 2025-02-12 152523](https://github.com/user-attachments/assets/d27d45d3-2552-4532-bb10-4249abdad359)

The `forge fmt --check` might run into error as it verifies the formatting, putting the `forge fmt` before the check makes it so that it formats the code first.
Hope this fix this error, as it did for me 😃

**Reply by EngrPips** [reply]
Great heads-up.

**Reply by yijiale-1120** [reply]
thank you friend，I just encountered the same error and solved it with your method.

**Reply by yijiale-1120** [reply]
But then it seems that the ctrl+s formatting on vs code is not the same as the formatting required on git hub.

**Reply by EngrPips** [reply]
🤔
